vignettes/dettl.Rmd

```diff
@@ -358,37 +355,66 @@ This will rollback any changes which have been made to the database before the c
 options(testthat.default_reporter = default_reporter)
 ```
 
-## Automatic load stage {#dettl_auto_load}
 
-`dettl` has an automatic implementation for the load stage of the import. At its most simple this will just take the named list of `transformed_data` and append each of these data frames to the table in the database with matching name. To use the automatic load step requires defining that the automatic load should be used in the load block of `dettl.yml` e.g.
+## Other stuff
 
-```yml
+### I want to write an import which creates a table {#create_mode}
+
+If you want to create a table during your import you need to switch the import mode. This is done in the `dettl.yml` configuration
+
+```
+mode: create
+```
+
+Create mode imports will skip some of the checks append imports do e.g. checks that the transformed data matches tables in the DB schema. The automatic load can still be used and will create each of the tables in the transformed data. Note that create and appending to tables cannot be mixed within an import.
+
+### I want to write a custom load
+
+Dettl allows writing a custom load if the automatic load does not achieve desired effect. You might want to use a custom load if you want to run an `UPDATE` query on the database. To do this update the `dettl.yml` for the report to use a user defined load
+
+```yaml
 load:
-  automatic: TRUE
+  func: load
   verification_queries: test_queries
   test: R/test_load.R
 ```
 
-Using an automatic load means that the configured `sources` in the `dettl.yml` no longer need to define a `load` function. A file containing tests for the load stage is still required.
+then add a function matching the name specified in the config, in our example this is just `load` which takes the transformed data and a DB connection e.g.
 
-The automatic load will load data respecting primary key constraints which are read automatically from the database. For example if we want to load data where we have `transformed_data` taking the form
+```R
+load <- function(transformed_data, con) {
+  ## Do something
+}
+```
 
-```{r echo = FALSE}
-people <- data.frame(c(1,2),
-                     c("Alice", "Bob"),
-                     c(25, 45),
-                     c(175, 187),
-                     stringsAsFactors = FALSE)
-colnames(people) <- c("id", "name", "age", "height")
-jobs <- data.frame(c(1, 2),
-                   c("researcher", "developer"),
-                   stringsAsFactors = FALSE)
-colnames(jobs) <- c("person", "job")
-knitr::kable(people, caption = "people")
-knitr::kable(jobs, caption = "jobs")
+Note that the containing file must be in the list of sources for the import.
+Expect that this stage is only really a set of update queries to be run on the database implemented in R.
+
```

Comment:
It might be good to say that if you use a custom load, then be careful not to specify numerical ids where the database would normally do that with a serial. Either leave the ids out and let the database do that itself, or if you require those ids in another place, then use an sql query with "RETURNING id" to fetch back the assigned serials. The key point is that doing a non-automatic custom load means dettl won't automatically fix your keys for you.... (and if you get this wrong like I did, you break the database!)